### PR TITLE
feat: Azure Functions v4 support

### PR DESCRIPTION
**Pull request by ctcampbell**
This pull request adds a new function `azureFunctionV4` to support Azure Functions v4.

Main interface changes:

* <a href="diffhunk://#diff-5c2ee26ede6e30cf7ddb3fc9441d269f1b706373c1084338571b033788d5cc34R1-R21">`azure-function-v4.js`</a>: Added a new function `azureFunctionV4` to handle incoming webhook requests from GitHub and verify their authenticity using `probot.webhooks.verifyAndReceive`.
* <a href="diffhunk://#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R3-R5">`index.js`</a>: Added a new function `createAzureFunctionV4`. <a href="diffhunk://#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R3-R5">[1]</a>

Documentation changes:

* <a href="diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L21-R45">`README.md`</a>: Added instructions for using `createAzureFunctionV4` function, including an example of creating an Azure Function using `@probot/adapter-azure-functions` package.

### CODE REVIEW

File: tsconfig.json
Comment (gr2m) [inline tsconfig.json:1]: do we need a tsconfig file given that this is a JS project? Is `index.d.ts` generated or did you hand craft it? If we need it, shouldn't it be a `jsconfig.json` file instead?

File: azure-function-v4.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @param {import('probot').Probot} probot
+ * @param {import('@azure/functions').HttpRequest} request
+ * @param {import('@azure/functions').InvocationContext} context
+ * @returns {Promise<import('@azure/functions').HttpResponseInit>}
+ */
+async function azureFunctionV4(probot, request, context) {
+  await probot.webhooks.verifyAndReceive({
+    id: request.headers.get("X-GitHub-Delivery"),
+    name: request.headers.get("X-GitHub-Event"),
+    signature: request.headers.get("X-Hub-Signature-256") || request.headers.get("X-Hub-Signature"),
```
Comment (gr2m) [inline azure-function-v4.js:13]: I don't think there is any GHES version left that sends out the `X-Hub-Signature` header, I think we can just leave it out.

File: tsconfig.json
Comment (ctcampbell) [reply tsconfig.json:1]: The `index.d.ts` file was generated, following docs at https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html, which describe adding `tsconfig.json`. I did try `jsconfig.json` but using that creates the following error:
image

File: azure-function-v4.js (hunk 1 above)
Comment (ctcampbell) [reply azure-function-v4.js:13]: Removed.

**Review by gr2m — APPROVED** [review]
Thank you 💐

**Pull request merged**